sklearn/tree/tests/test_tree.py

```diff
@@ -670,6 +670,30 @@ def check_min_weight_fraction_leaf(name, datasets, sparse=False):
             "min_weight_fraction_leaf={1}".format(
                 name, est.min_weight_fraction_leaf))
 
+    # test case with no weights passed in
+    total_weight = X.shape[0]
+
+    for max_leaf_nodes, frac in product((None, 1000), np.linspace(0, 0.5, 6)):
```

Comment:
What is the duration of this test? If it's more than a couple hundred milliseconds, please reduce to `np.linspace(0, 0.5, 3)`.